lib/waistband/index.rb

```diff
@@ -132,6 +132,40 @@ def save(*args)
       false
     end
 
+    def update!(*args)
+      check_permission!('write')
+
+      body_hash = args.extract_options!
+      id = args.first
+      _type = body_hash.delete(:_type) || body_hash.delete('_type') || default_type_name
+
+      # map everything to strings if need be
+      body_hash = stringify_all(body_hash) if @stringify
+
+      verify_body_size(config_name, _type, id, body_hash)
+
+      body_hash = { doc: body_hash }
```

Comment:
Most of this method is the same as `save`, but we need to wrap the hash in a special `doc` key as [explained here](https://www.elastic.co/guide/en/elasticsearch/reference/6.5/docs-update.html#_updates_with_a_partial_document). Could also potentially DRY up what `save` and `update` are both doing, but feel its a bit early to do that just yet.